src/SwitchInput.cpp

```diff
@@ -426,64 +426,117 @@ int AbstractHwRotaryEncoder::amountFromChange(unsigned long change) {
 }
 
 void HardwareRotaryEncoder::encoderChanged() {
-	bool lastSyncStatus = switches.getIoAbstraction()->sync();
-    bitWrite(flags, LAST_SYNC_STATUS, lastSyncStatus);
+    static uint8_t state = 0;        // Current state of the encoder
+    static uint8_t pulseCounter = 0; // Pulse counter for FULL_CYCLE and HALF_CYCLE modes
```

Comment:
One point, can we move these two to be fields in the class. Simply move them into `SwitchInput.h` inside the `private:` section of `HardwareRotaryEncoder`. 

```
class HardwareRotaryEncoder : public AbstractHwRotaryEncoder {
private:
    uint8_t aLast;
    uint8_t cleanFromB;
    uint8_t state = 0;
    uint8_t pulseCounter = 0;
    // rest of code
```

Otherwise feel free to commit and I'll move these afterwards to allow for more than one encoder at once..

Reply:
Of course we can do that.